ClientProxy.go

```diff
@@ -59,14 +65,22 @@ func (this ClientProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 		return
 	}
 	var requestBody []byte
+	//_, err = resp.Body.Read(requestBody)
 	nRead, err := resp.Body.Read(requestBody)
-	if err != nil || nRead < (int)(resp.ContentLength) {
+	if err != nil {
 		// these need to be separate checks, otherwise you will get a nil-reference
-                // when you print the error message below!
+		// when you print the error message below!
 		SRVFAIL(w, request)
 		_D("error in reading HTTP response, error message: %s", err)
 		return
 	}
+	//I not sure whether I should return server fail directly
```

Comment:
I think that if we use the `ReadAll()` function that we don't need to worry about this case. I hope. ;)